lib/classes/Variables.test.js

```diff
@@ -76,6 +74,33 @@ describe('Variables', () => {
         expect(serverless.service.resources.bar).to.equal(barValue);
       });
     });
+
+    it('should delete null environment variables', () => {
+      const serverless = new Serverless();
+
+      serverless.service.provider.environment = {
+        SOME_VAR: 'some-value',
+        OTHER_VAR: null,
+      };
+      serverless.service.functions = {
+        hello: {
+          handler: 'hello.js',
+          environment: {
+            VAR_ONE: 'one',
+            VAR_TWO: null,
+          },
+        },
+      };
+
+      return serverless.variables.populateService().then(() => {
+        /* eslint-disable no-unused-expressions */
```

Comment:
@pmuens I like the idea of disabling the test for the unit tests. When using chai, there are a lot of cases where you call a property to test for a specific value.
Alternatively we could use 
```
.then(() => BbPromise.all([
  expect(...).to.be.undefined,
  expect(...).to.be.not.empty,
  ...
]));
```
That way the warning can be enabled and it will additionally work with expectations that return promises or values.